manifests/istio-control/istio-discovery/templates/deployment.yaml

```diff
@@ -133,7 +133,11 @@ spec:
           - name: PILOT_ENABLE_PROTOCOL_SNIFFING_FOR_INBOUND
             value: "{{ .Values.pilot.enableProtocolSniffingForInbound }}"
           - name: INJECTION_WEBHOOK_CONFIG_NAME
+{{- if eq .Release.Namespace "istio-system"}}
             value: istio-sidecar-injector{{- if not (eq .Values.revision "") }}-{{ .Values.revision }}{{- end }}
```

Comment:
We should probably not add a namespace suffix to the webhook at all now that we have the revision suffix. I would prefer we make that change than this one

Reply:
I wasn't really sure which side should be "fixed". I can look into updating with that instead.

Reply:
Yeah I think back in the day we were allowing multiple control planes in a cluster based on namespace, but now we are doing it on revisions, so its more of a legacy artifact

Reply:
Does the test make sense in its current form or something similar? I've ran into several issues related to using a non-default namespace, like #22445.  I'm hoping to have some add some testing that seems missing for that.

I wasn't really sure what to put for the `diffSelect`.